scipy/signal/signaltools.py

```diff
@@ -2525,6 +2525,8 @@ def detrend(data, axis=-1, type='linear', bp=0):
         A sequence of break points. If given, an individual linear fit is
         performed for each part of `data` between two break points.
         Break points are specified as indices into `data`.
+    copy : boolean, optional
+        If False, try to avoid a copy and do inplace detrending instead. This may reduce memory usage. Default true
```

Comment:
Can you PEP8-ify this with < 80 char lines?

Reply:
Done.

I did what ewmoore suggested and changed the name from "copy" to "overwrite_data." This also meant swapping the logic around, so copy=True equates to overwrite_data=False.

I've also added in a couple of simple unit tests, and one that directly compares overwrite_data=True and overwrite_data=False.